lib/diagnostic.rb

```diff
@@ -43,6 +44,10 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+      lines = File.open(filename[0]) do |file|
+      file.each_with_object(lines: 0, new_lines: 0) do |line, memo|
+        memo[:lines] += line.split()
+        memo[:new_lines] += line.join(' ')
+      end
```

Comment:
To load a file, you run `File.open(filename)`. This gives you a new File object. The File class includes the Enumerable module, so Files have all of the Enumerable methods, such as `.map`. `.map` accepts a block, and passes that block a single item (in this case, a line of text from the file) as it iterates through the collection.